docs/DIDYOUKNOW.md

```diff
@@ -0,0 +1,195 @@
+# Did You Know?
+
+## Language
+
+Did you know that:
+
+- Flix offers a unique combination of features, including: algebraic data types and pattern matching, 
+  extensible records, type classes, higher-kinded types, channel and process-based concurrency, 
+  a polymorphic effect system, purity polymorphic functions, and first-class Datalog constraints.
+
+- Flix has no implicit global state. Any global state must be passed around explicitly.
+
+- Flix is one language. There are no pragmas or compiler flags to enable or disable features. 
+
+- Flix supports type parameter elision. That is, in most cases, a polymorphic function does 
+  can be written without explicitly declaring its type parameters.
+
+- the Flix type and effect system can enforce purity.
+
+- Flix supports effect polymorphism? For example, the effect of `List.map` 
+  depends on the effect of its argument.
+
+- every declaration in Flix is private by default.
+
+- In Flix nothing in is executed before main. There is no global state nor any static initializers.
+
+- Flix supports full tail call elimination, i.e. tail calls do not grow the stack.
+  (Flix-- being on the JVM -- has to emulate tail calls until Project Loom arrives.)
+
+- Flix supports extensible records with row polymorphism.
+
+- Flix supports string interpolation by default, e.g. "Hello ${name}". String interpolation
+  uses the `ToString` type class -- as it should be.
+
+- Flix supports the "pipeline" operator `|>` and the Flix standard library is designed 
+  to make good use of it.
+
+- In Flix variables and function definitions must be lowercase whereas enum cases and types must be uppercase. 
+  This convention makes it faster and simpler to read signatures.
+
+- Flix supports set and map literals `Set#{1, 2, 3}` and `Map#{1 => 2, 3 => 4}`.
+
+- Flix supports monadic do-notation called `let*`.
+
+- Flix supports "program holes" written as either `???` or as `?name`.
+
+- Flix supports infix function applications via backticks.
+
+- Flix compiles to JVM bytecode and runs on the Java Virtual Machine.
+
+- Flix supports channels and processes, including the powerful `select` expression.
+
+- Flix supports first-class Datalog constraints, i.e. Datalog program fragments are values.
+
+- Flix supports compile-time checked stratified negation.
+
+- Flix supports partial application, i.e. a function can be called with fewer
+  arguments that its declared number of formal parameters.
+
+- the Flix type and effect system is based on Hindley-Miler.
+  The same core type system that powers OCaml, Standard ML, and Haskell.
+
+- the Flix type and effect system is sound, i.e. if a program type checks
+  then a type error cannot occur at run-time.
+
+- the Flix type and effect system supports complete type inference, i.e.
```

Comment:
Is there a caveat here about ambiguous signatures that require annotations?

Reply:
Yes, but then again, we also require top-level type signatures.